### ISSUE TITLE
Need fix for broken prediction

### ISSUE THREAD

**Issue by muazhari** [zemlyansky/arima#7]
I tried to train/fit ARIMA & Auto ARIMA model with sin pattern datasets but it gives an invalid prediction, even in the first data point is invalid/not meaningful at all. can I get the solution or someone helps me out? I truly need this fix.

```
  const model = new ARIMA({
    auto: true,
    verbose: true,
  });

  model.train(trainDatasets);

  const [preds, errors] = model.predict(testLen);
```

below is the plotted data.

![image](https://user-images.githubusercontent.com/39398937/110334613-8c56b400-8055-11eb-82ab-7238d19be231.png)

**Comment by ghost** [comment]
It's issue with auto mode. The api in library is created based on sarimmax function. There are few bugs with the parameters estimation. For your usage you can use the arima with p:2, d:1, q:2 parameters. It should produce the meaningfull result if no, add the seasonal `s` parameter. You need to count it by your own. It could be 16 or 8. This is the only one solution for you currently.

**Comment by zemlyansky** [comment]
Yep, it seems that there's some issue with the AutoARIMA mode. Can be related to [https://github.com/rafat/ctsa/issues/5](https://github.com/rafat/ctsa/issues/5)
